### PR TITLE
CLI dbt: Add support for linked tables

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-106

Related `go-client` PR: https://github.com/keboola/go-client/pull/80

Changes:
- All CLI dbt commands works with linked buckets.

-----------------

Manualne testovanie, kedze nemame API requests na vytvorenie linked buckets + pre testovanie nad viacerými projektmi.

image

image

image

### CODE REVIEW

File: pkg/lib/operation/dbt/generate/env/operation.go [hunk 1]
```diff
@@ -33,23 +36,54 @@ func Run(ctx context.Context, opts GenerateEnvOptions, d dependencies) (err erro
...
+
+	if len(linkedBucketEnvsMap) > 0 {
+		var linkedBucketEnvs []string
+		for env := range linkedBucketEnvsMap {
+			linkedBucketEnvs = append(linkedBucketEnvs, env)
+		}
+		l.Info()
+		l.Info("Note:")
+		l.Info("  The project contains linked buckets that are shared from other projects.")
+		l.Info("  Each shared project has a different database, so additional environment variables")
```
Comment (michaljurecko) [inline pkg/lib/operation/dbt/generate/env/operation.go:84]: Fisa tu chcel nejaku poznamku.

File: pkg/lib/operation/dbt/generate/sources/generate_test.go [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+						Quoting: dbt.SourceTableQuoting{
+							Database:   true,
+							Schema:     true,
+							Identifier: true,
+						},
+					},
+				},
+			},
+		},
+	}, res)
```
Comment (michaljurecko) [inline pkg/lib/operation/dbt/generate/sources/generate_test.go:209]: Doplneny unit test pre sources.

File: pkg/lib/operation/dbt/listbuckets/group_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+			DatabaseEnv: "DBT_KBC_TARGET1_DATABASE",
+			Tables:      []keboola.Table{secTable1, secTable2},
+		},
+		{
+			LinkedProjectID: 12345,
+			SourceName:      linkedBucket.ID.String(), // defined by the linked bucket name
+			Schema:          "out.c-shared",           // defined by the source bucket name
+			DatabaseEnv:     "DBT_KBC_TARGET1_12345_DATABASE",
+			Tables:          []keboola.Table{linkedTable1, linkedTable2},
+		},
```
Comment (michaljurecko) [inline pkg/lib/operation/dbt/listbuckets/group_test.go:100]: Doplneny unit test, `listbuckets` je vrstva ktora nacita a zgroupuje tables by bucket.
Teda pripravi data pre `sources` a `envs`.

Predty sa to robilo iba v `sources`, ale teraz potrebujem `buckets` aj v `envs`, takze som to musel upravit.

**Review by JakubMatejka — APPROVED** [review]
(hodláš to mergovat až po mergi a release toho PR v go-klientovi, jo? 😄)

**Comment by michaljurecko** [comment]
jj

**Pull request merged**